bin/agent-data-plane/src/main.rs

```diff
@@ -174,8 +175,14 @@ fn create_topology(
         .with_transform_builder(host_enrichment_config)
         .with_transform_builder(origin_enrichment_config);
     let internal_metrics_remap_config = AgentTelemetryRemapperConfiguration::new();
-    let dd_metrics_config = DatadogMetricsConfiguration::from_configuration(configuration)
+
+    let config_refresher_configuration = RefresherConfiguration::from_configuration(&configuration)?;
+    let config_refresher: Arc<RefreshableConfiguration> = Arc::new(config_refresher_configuration.build()?);
+    config_refresher.clone().spawn_refresh_task();
```

Comment:
I think we could make this less clunky by having `RefresherConfiguration::build` do the spawning for us, taking care of cloning `RefreshableConfiguration`.

This will necessitate needing to change `values` in `RefreshableConfiguration` from `ArcSwap<Value>` to `Arc<ArcSwap<Value>>` (yes, I realize the irony of having to wrap `ArcSwap` in an `Arc` to make it shareable 😄) and adding a derive for `Clone`, but it'll mean we just have callers deal with `RefreshableConfiguration`.

Reply:
> This will necessitate needing to change values in RefreshableConfiguration from ArcSwap<Value> to Arc<ArcSwap<Value>>

Can you explain more on why it needs to be wrapped in an `Arc` if the overall struct is in an `Arc`?